src/Workspaces/Core/Portable/Workspace/Workspace_Events.cs

```diff
@@ -42,6 +43,23 @@ public event EventHandler<WorkspaceChangeEventArgs> WorkspaceChanged
         }
     }
 
+    /// <summary>
+    /// An event raised *immediately* whenever the current solution is changed. Handlers
+    /// should be written to be very fast.
```

Comment:
```suggestion
    /// An event raised *immediately* whenever the current solution is changed. Handlers
    /// should be written to be very fast. Called on the same thread changing the workspace,
    /// which may vary depending on the workspace.
```

Just to make it clear which thread it's on (and that it's probably not great to depend on that).